### DISCUSSION TITLE
The `@eslint/js` dependency

### DISCUSSION THREAD

**Discussion by RobinTail** [eslint/eslint#18721] [Ideas] [answered]
The recommended installation command `npm init @eslint/config@latest` from the "Quick start" guide

![image](https://github.com/user-attachments/assets/3abdc405-c4ed-4812-ad0f-95cdda917f1f)

installs `eslint` itself, but also `@eslint/js`, while `@eslint/js` is listed as a dependency of `eslint`, moreover having strict version:

https://github.com/eslint/eslint/blob/63881dc11299aba1d0960747c199a4cf48d6b9c8/package.json#L73

This leads to a possibility of two different versions of `@eslint/js` being installed. Here is the `package.json` after installation:

```json
{
  "name": "test",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "devDependencies": {
    "@eslint/js": "^9.8.0",  // <—— notice caret
    "eslint": "9.x",
    "globals": "^15.8.0"
  }
}
```

- If `@eslint/js` is expected to be installed by consumer (for using in config), maybe it should be a `peerDependency` instead?
- If consumer is not expected to use the version of `@eslint/js` out of sync with `eslint`, then maybe it should not be installed in cosumer `package.json` and exposed through `eslint` itself?

Any thoughts on that or enlighten me please if I'm missing something here.

**Comment by RobinTail** [comment]
CC @fasttime

**Reply by fasttime** [reply, marked as answer]
`@eslint/js` provides the configurations `recommended` and `all` for use by consumers. Since your `package.json` file was created using `npm init @eslint/config@latest`, I would expect `@eslint/js` to be imported as a dependency in the config file created by the same command. It seems correct to have it in `devDependencies` in that case I think.

It is also possible in this setup that the version of `eslint` lies behind `@eslint/js`. I agree this is suboptimal, but that should not be a problem in most cases, because breaking changes are typically deferred to major releases, while both `^9.8.0` and `9.x` would install versions in the v9 release line.

**Reply by RobinTail** [reply]
okay
